src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Emitters/EndpointParameterEmitter.cs

```diff
@@ -291,7 +291,7 @@ internal static void EmitServiceParameterPreparation(this EndpointParameter endp
     public static string EmitArgument(this EndpointParameter endpointParameter) => endpointParameter.Source switch
     {
         EndpointParameterSource.JsonBody or EndpointParameterSource.Route or EndpointParameterSource.RouteOrQuery or EndpointParameterSource.JsonBodyOrService or EndpointParameterSource.FormBody => endpointParameter.IsOptional ? endpointParameter.EmitHandlerArgument() : $"{endpointParameter.EmitHandlerArgument()}!",
-        EndpointParameterSource.Unknown => throw new Exception("Unreachable!"),
+        EndpointParameterSource.Unknown => throw new InvalidOperationException($"Unexpected {nameof(EndpointParameterSource)}: {endpointParameter.Source}"),
```

Comment:
Can we stick with `NotImplementedException` like you did for the other "Unreachable!" cases? I also think the message should still start with "Unreachable!" for documentation purposes assuming it really is unreachable.

```suggestion
        EndpointParameterSource.Unknown => throw new NotImplementedException($"Unreachable! Unexpected {nameof(EndpointParameterSource)}: {endpointParameter.Source}"),
```